osu.Game/Rulesets/UI/ModIcon.cs

```diff
@@ -74,11 +49,33 @@ public ModIcon()
                     Origin = Anchor.Centre,
                     Anchor = Anchor.Centre,
                     Colour = OsuColour.Gray(84),
-                    TextSize = 20
+                    TextSize = 20,
+                    Icon = mod.Icon
                 },
             };
 
             reapplySize();
         }
+
+        private void reapplySize()
+        {
+            background.TextSize = icon_size;
+            modIcon.TextSize = icon_size - 35;
+        }
+
+        private Color4 getBackgroundColourFromMod(Mod mod)
+        {
+            switch (mod.Type)
+            {
+                case ModType.DifficultyIncrease:
+                    return OsuColour.FromHex(@"ffcc22");
+                case ModType.DifficultyReduction:
+                    return OsuColour.FromHex(@"88b300");
+                case ModType.Special:
+                    return OsuColour.FromHex(@"66ccff");
+
+                default: return Color4.White;

```

Comment:
This is a weird structure to this default. Please follow the case statement structure (incl. removing the newline before it).